www/profile/index.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+$servername = "localhost";
+$username = "worker";
+$password = "swiffcat";
+$db = "socialmedia";
+
+// Create connection
+$conn = new mysqli($servername, $username, $password, $db);
+
+// Check connection
+if ($conn->connect_error) {
+    header("HTTP/1.1 500 Internal Server Error");
+    die();
+}
+
+$headers = getallheaders();
+if (!array_key_exists("token", $headers)) {
+    header("HTTP/1.1 401 Unauthorized");
+    die();
+}
+$token = getallheaders()["token"];
+
+// If invalid header token, die
+$stmt = $conn->prepare("SELECT user_id FROM auth WHERE token = ?");
+$stmt->bind_param("s", $a1);
+$a1 = $token;
+$stmt->execute();
+$stmt->store_result();
+$accessor_id = null;
+$stmt->bind_result($accessor_id);
+if ($stmt->fetch()) {
+    // user_id is bound
+} else {
+    header("HTTP/1.1 401 Unauthorized");
+    die();
+}
+
+if ($_SERVER['REQUEST_METHOD'] === 'GET') {
+    if (!isset($_GET["id"])) {
+        echo $_GET;
+        header("HTTP/1.1 403 Forbidden");
+        die();
+    }
+
+    $user_id = $_GET["id"];
+
+    // make sure the user exists and grab their data
+    $stmt = $conn->prepare("SELECT * FROM users WHERE user_id = ?");
+    $stmt->bind_param("s", $a1);
+    $a1 = $user_id;
+    $stmt->execute();
+    $result = $stmt->get_result();
+    $count = $result->num_rows;
+    if ($count != 1) {
+        header("HTTP/1.1 403 Forbidden");
+        die();
+    }
+    $row = $result->fetch_assoc();
+
+    // Retrieve data from the posts table
+    $stmt = $conn->prepare("SELECT * FROM posts WHERE user_id = ? ORDER BY created_at DESC");
+    $stmt->bind_param("s", $a1);
+    $a1 = $user_id;
+    $stmt->execute();
+    $result = $stmt->get_result();
+    $posts = array();
+
+    if ($result->num_rows > 0) {
+        while ($post_row = $result->fetch_assoc()) {
+            $posts[] = $post_row;
+        }
+    }
+
+    // add posts to the values obtained earlier
+    $row["posts"] = $posts;
+    $json = json_encode($row);
+    echo $json;
+
+    // Close the database connection
+    mysqli_close($conn);
+} else if ($_SERVER['REQUEST_METHOD'] === 'PUT') {
+
+    $inputData = file_get_contents('php://input');
+
+    // Split the input into parts
+    $boundary = substr($inputData, 0, strpos($inputData, "\r\n"));
```

Comment:
Where's the error checking?
Also \r\n is a windows newline sequence, not a unix or linux one